src/blocked.c

```diff
@@ -545,13 +531,8 @@ void blockForKeys(client *c, int btype, robj **keys, int numkeys, mstime_t timeo
 
     c->bstate.timeout = timeout;
     for (j = 0; j < numkeys; j++) {
-        /* Allocate our bkinfo structure, associated to each key the client
-         * is blocked for. */
-        bkinfo *bki = zmalloc(sizeof(*bki));
-
         /* If the key already exists in the dictionary ignore it. */
-        if (dictAdd(c->bstate.keys,keys[j],bki) != DICT_OK) {
-            zfree(bki);
+        if (dictFetchValue(c->bstate.keys,keys[j])) {
```

Comment:
i think we can use dictAddRaw and later conditionally update the dictEntry, instead of doing double lookup (find and add)

Reply:
done